### PR TITLE
Support custom oauth client id and redirect port

### PR DESCRIPTION
**Pull request by jackyhu-db**
PySQL is used by other tools/CLIs which have their own oauth client id, we need to expose `oauth_client_id` and `oauth_redirect_port` as the connection parameters to support this customization.

### CODE REVIEW

File: src/databricks/sql/auth/auth.py [hunk 1]
```diff
@@ -89,8 +89,9 @@ def get_python_sql_connector_auth_provider(hostname: str, **kwargs):
         use_cert_as_auth=kwargs.get("_use_cert_as_auth"),
         tls_client_cert_file=kwargs.get("_tls_client_cert_file"),
         oauth_scopes=PYSQL_OAUTH_SCOPES,
-        oauth_client_id=PYSQL_OAUTH_CLIENT_ID,
-        oauth_redirect_port_range=PYSQL_OAUTH_REDIRECT_PORT_RANGE,
+        oauth_client_id=kwargs.get("oauth_client_id") or PYSQL_OAUTH_CLIENT_ID,
+        oauth_redirect_port_range=kwargs.get("oauth_redirect_port_range") if kwargs.get(
```
[the remark this answers is not in the document]
Comment (jackyhu-db) [reply src/databricks/sql/auth/auth.py:93]: Hi @moderakh good point. It makes sense, will change it.

[the remark this answers is not in the document]
Comment (susodapop) [reply src/databricks/sql/auth/auth.py:93]: This doesn't feel like a very pythonic approach. If we want to configure using an object then we should just use a `OAuthConfig` object with a property for each element. There's no problem having dozens of available kwargs. In fact, that's what `kwargs` already _is_: a dictionary that contains extra stuff. I think changing this is an unnecessary extra step that is neither obvious to users nor necessary.

[the remark this answers is not in the document]
Comment (susodapop) [reply src/databricks/sql/auth/auth.py:93]: Case in point, there are already dozens of other kwargs and we aren't going to organise them into "sub-kwargs". It's fine to have these at the top-level of the kwargs namespace.

**Review by susodapop — APPROVED** [review]

**Review by moderakh — APPROVED** [review]
LGTM. thanks @jackyhu-db

**Pull request merged**